app/controllers/application_controller.rb

```diff
@@ -69,6 +69,6 @@ def user_locale
   end
 
   def http_head_locale
-    http_accept_language.compatible_language_from(I18n.available_locales)
+    http_accept_language.language_region_compatible_from(I18n.available_locales)
```

Comment:
`compatible_language_from` would ignore language region from `Accept-Language` and choose the first compatible language. For example:

```
  http_accept_language.user_preferred_languages # => %w(zh-TW)
  available = %w(zh-CN zh-TW)
  http_accept_language.compatible_language_from(available) # => 'zh-CN'

  http_accept_language.language_region_compatible_from(available) # => 'zh-TW'
```